zebra-chain/src/transaction/unmined.rs

```diff
@@ -0,0 +1,171 @@
+//! Unmined Zcash transaction identifiers and transactions.
+//!
+//! Transaction versions 1-4 are uniquely identified by narrow transaction IDs,
+//! whether they have been mined or not,
+//! so Zebra and the Zcash network protocol don't use wide transaction IDs for them.
+//!
+//! Zebra's [`UnminedTxId`] and [`UnminedTx`] enums provide the correct unique ID for
+//! unmined transactions. They can be used to handle transactions regardless of version,
+//! and get the [`WtxId`] or [`Hash`] when required.
+
+use std::sync::Arc;
+
+#[cfg(any(test, feature = "proptest-impl"))]
+use proptest_derive::Arbitrary;
+
+use super::{
+    AuthDigest, Hash,
+    Transaction::{self, *},
+    WtxId,
+};
+
+use UnminedTxId::*;
```

Comment:
Nit: Would it make sense to move the wildcard imports closer to where they are used (even if it becomes duplicate)? It's a bit harder to find where something is defined if it's imported through a wildcard, so keeping them close to where they are used limits the searches.

Reply:
In general, I don't mind either way.

But my install of `rust-analyzer` fully qualifies enum wildcard imports if the import is in the method.
So I tend to put it near the top of the file.